cmd/fossa/main.go

```diff
@@ -115,136 +114,44 @@ func main() {
 	app.Run(os.Args)
 }
 
-func setupModule(config moduleConfig, manifestName string, moduleType module.Type) (module.Module, error) {
+func resolveModuleConfig(moduleConfig config.ModuleConfig) (builders.Builder, module.Module, error) {
+	mainLogger.Debugf("Resolving ModuleConfig: %#v", moduleConfig)
+
+	var builder builders.Builder
 	var m module.Module
+	var err error
 
-	modulePath, err := filepath.Abs(config.Path)
-	if filepath.Base(modulePath) == manifestName {
-		modulePath = filepath.Dir(modulePath)
-	}
-	if err != nil {
-		return m, err
+	moduleType := config.GetModuleType(moduleConfig.Type)
+	if moduleType == "" {
+		mainLogger.Debug("Got unknown module.")
+		return builder, m, fmt.Errorf("unknown module type: %s", moduleConfig.Type)
 	}
 
-	moduleName := config.Name
-	if moduleName == "" {
-		moduleName = config.Path
-	}
+	mainLogger.Debugf("Got <%s> module.", moduleType)
```

Comment:
You don't need to surround `%s` with brackets. If you feel like you need to, you should probably be using `%#v` instead.